cmd/start.go

```diff
@@ -70,11 +77,23 @@ $ %s start demo-path2 --max-tx-size 10`, appName, appName)),
 				return err
 			}
 
+			timeInterval := viper.GetDuration(flagTimeInterval)
+
+			go func() {
+				for {
+					// run for every 60 seconds
+					if err := UpdateClientsFromChains(c[src], c[dst]); err != nil {
+						fmt.Printf("Error in updating clients....%v", err)
+					}
+					time.Sleep(timeInterval)
```

Comment:
Why are we sleeping at a uniform interval? This is unnecessary and wasteful imo. We can already find out when the next client needs to be updated using the above `UpdateClientsFromChains` function. We can simply look at which client is closer to expiring and return a `timeToNextExpiry` in `UpdateClientsFromChains`. This command can then just sleep form `timeToNextExpiry - ThresholdTime`

Reply:
@akhilkumarpilli lets do this. I realize now that the client will likely be updated while sleeping (packets being relayed cause updates) so even after waking, the client may not be within the update threshold. 

Let's keep the threshold flag and remove the time interval flag. 

So the flow should be, wakeup/start, check expiry time, if within the threshold, update, else return the `timeToNextExpiry - ThresholdTime` and sleep on that